processor/samplingprocessor/tailsamplingprocessor/processor_test.go

```diff
@@ -331,6 +331,54 @@ func TestSamplingPolicyDecisionNotSampled(t *testing.T) {
 	require.Equal(t, 2, mpe.LateArrivingSpansCount, "policy was not notified of the late span")
 }
 
+func TestMultipleBatchesAreCombinedIntoOne(t *testing.T) {
+	const maxSize = 100
+	const decisionWaitSeconds = 1
+	// For this test explicitly control the timer calls and batcher, and set a mock
+	// sampling policy evaluator.
+	msp := new(exportertest.SinkTraceExporter)
+	mpe := &mockPolicyEvaluator{}
+	mtt := &manualTTicker{}
+	tsp := &tailSamplingSpanProcessor{
+		ctx:             context.Background(),
+		nextConsumer:    msp,
+		maxNumTraces:    maxSize,
+		logger:          zap.NewNop(),
+		decisionBatcher: newSyncIDBatcher(decisionWaitSeconds),
+		policies:        []*Policy{{Name: "mock-policy", Evaluator: mpe, ctx: context.TODO()}},
+		deleteChan:      make(chan traceKey, maxSize),
+		policyTicker:    mtt,
+	}
+
+	mpe.NextDecision = sampling.Sampled
+
+	traceIds, batches := generateIdsAndBatches(3)
+	for _, batch := range batches {
+		require.NoError(t, tsp.ConsumeTraces(context.Background(), batch))
+	}
+
+	tsp.samplingPolicyOnTick()
+	tsp.samplingPolicyOnTick()
+
+	require.EqualValues(t, 3, len(msp.AllTraces()), "There should be three batches, one for each trace")
+
+	receivedTraces := msp.AllTraces()
+	for i, traceID := range traceIds {
+		trace := receivedTraces[findTrace(receivedTraces, traceID)]
+		require.EqualValues(t, i+1, trace.SpanCount(), "The trace should have all of its spans in a single batch")
+	}
+}
+
+func findTrace(a []pdata.Traces, traceID pdata.TraceID) int {
```

Comment:
The name is a bit misleading, as it's not finding a trace, it's returning the number of traces that were found for the given ID. Suggestion: `numTracesWithID()`?

Reply:
There is probably a better way to do this, but its finding the Index of the trace in the array. I added this because there isn't a guaranteed order in which the traces are received 

In the case where the trace is not found it returns the size of the array which might be confusing, I could change this to `-1` if you'd like.

Reply:
Would using `t.Fatal()` or `assert.Fail()` be more appropriate since it's invalid test state?

Reply:
I've updated this method to return the trace or nil which makes it much clearer